src/v/storage/segment_utils.cc

```diff
@@ -713,8 +712,9 @@ ss::future<> write_concatenated_compacted_index(
       std::move(segments),
       [cfg, target_path = std::move(target_path)](
         std::vector<ss::lw_shared_ptr<segment>>& segments) mutable {
+          auto compacted_path = internal::compacted_index_path(target_path);
           return do_write_concatenated_compacted_index(
-            std::move(target_path), segments, cfg);
+            std::move(compacted_path), segments, cfg);
       });
```

Comment:
can you add some information to the commit message about what is being fixed here? it may be fixing a bug, but if it is, it looks like other code would have needed to be changed as well and that isn't clear.

for example: the caller of this method looks like it passes a staging index file as the target path and then expects that after this method completes that the staging index index path exists and operates on it. this patch seems to circumvent this and write into a different file than the caller would expect.

also, target_path is unused in the lambda now.

Reply:
you are right, I've fixed it by passing compacted index file path at the call site.